src/core/Akka/Configuration/FlatConfig.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Text;
+using Hocon;
+
+namespace Akka.Configuration
+{
+    public class FlatConfig:Config
+    {
+        private ConcurrentDictionary<string, HoconValue> _cache = new ConcurrentDictionary<string, HoconValue>();
+
+        public FlatConfig(HoconRoot root, Config fallback) : base(root, fallback)
+        {
+            if (root is FlatConfig caching)
+                _cache = caching._cache;
+            else
+                _cache = new ConcurrentDictionary<string, HoconValue>();
+        }
+
+        public FlatConfig(HoconRoot root) : base(root)
+        {
+            if (root is FlatConfig caching)
+                _cache = caching._cache;
+            else
+                _cache = new ConcurrentDictionary<string, HoconValue>();
+        }
+
+        protected override HoconValue GetNode(string path)
+        {
+            if (_cache.TryGetValue(path, out var result))
+                return result;
+            result = base.GetNode(path);
+            _cache[path] = result;
+            return result;
+        }
+
+        protected override HoconValue GetNode(HoconPath path)
+        {
+            var fullPath = path.ToString();
+            if (_cache.TryGetValue(fullPath, out var result))
+                return result;
+            result = base.GetNode(path);
+            _cache[fullPath] = result;
```

Comment:
So we shouldn't do this - it'll create a large memory footprint for systems that have to deploy a huge number of actors. We want to avoid storing a cached object for _every single actorpath_ we resolve. Ideally, we want to only store cached objects for paths that get used.

This is what the `CachingConfig` does - we have to strike a balance between memory and throughput here and this goes too far in the direction of trading away memory for throughput.

Reply:
Its not, really... it only caches the hits and ignores the misses, the memory hit shouldn't be that bad.

Reply:
Got it - based on our conversation from earlier it sounds like I misunderstood.